src/cpp/database/database.cpp

```diff
@@ -1142,6 +1142,19 @@ void Database::link_participant_with_process_nts(
 
     /* Add entry to processes_by_domain_ */
     processes_by_domain_[domain_id][process_it->first] = process_it->second;
+
+    // If the participant is active, the process must be active
+    if (participant->active)
+    {
+        if (!participant->process->active)
+        {
+            change_entity_status_of_kind(participant->process->id, true, EntityKind::PROCESS, domain_id);
+        }
+        else if (!participant->process->user->active)
+        {
+            change_entity_status_of_kind(participant->process->user->id, true, EntityKind::USER, domain_id);
+        }
```

Comment:
I have opted to create host/user/process as inactive always. 
I erased the test `link_participant_with_active_process_inactive_user` because this case now is impossible, Always when we link participant-process, the process is inactive and it will activate all the sub-entities if is necessary.